drivers: dmic: update dmic flow initialization

dnikodem opened the pull request:
DMIC configuration should allow the use of periodic start.

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

dnikodem committed:
drivers: dmic: update dmic flow initialization

DMIC configuration should allow the use of periodic start.

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

juimonen reviewed (dismissed):
lgtm

singalsu reviewed (commented):
Additions looks OK. Please check that the dmic_start/stop procedure with un-reset to start is compatible with this mode.

dnikodem committed:
drivers: dmic: remove soft_reset from dmic init flow

DMIC does not need to use SOFT_RESET bit to start collecting data if
periodic_start bit is enabled.

Signed-off-by: Damian Nikodem <damian.nikodem@intel.com>

singalsu commented on drivers/dai/intel/dmic/dmic_nhlt.c:
```diff
@@ -344,11 +345,28 @@ int dai_dmic_set_config_nhlt(struct dai_intel_dmic *dmic, const void *bespoke_cf
 		return -EINVAL;
 	}
 
+	const uint32_t skip_regs[] = { (0x8 / 4), (0x10 / 4), (0x14 / 4), (0x18 / 4),
```
Why is this done?

dnikodem replied:
I removed this part - latest change set MIC_MUTE bit for unused PDM registers only. Thanks

abonislawski reviewed (approved):
It was tested in dmic part of our full scope tests?

dnikodem commented:
> It was tested in dmic part of our full scope tests?

The previous version of this patch was tested in dmic range, I am waiting for the test results with the latest version. Thanks

dnikodem commented:
> > It was tested in dmic part of our full scope tests?
> 
> The previous version of this patch was tested in dmic full scope, I am waiting for the test results with the latest version. Thanks

According to the latest test results with newest patch, I do not observe any new regression in dmic scope.
